code/modules/projectiles/guns/energy/special.dm

```diff
@@ -25,6 +64,10 @@
 	cell_type = "/obj/item/weapon/cell/crap/better"
 	projectile_type = "/obj/item/projectile/ion/small"
 
+/obj/item/weapon/gun/energy/ionrifle/ioncarbine/attackby(var/obj/item/A as obj, mob/user as mob)
+	if(istype(A, /obj/item/tool/circular_saw) || istype(A, /obj/item/weapon/melee/energy) || istype(A, /obj/item/weapon/pickaxe/plasmacutter))
```

Comment:
instead of this, make the parent have a new proc/saw_off() that handles the sawing off, called from attackby

then just override saw_off() with a return in the chiild